src/library.js

```diff
@@ -2634,9 +2634,22 @@ LibraryManager.library = {
   // setjmp.h
   // ==========================================================================
 
+  // asm.js-style setjmp/longjmp support for wasm
```

Comment:
this first line suggests the code following is just for wasm. maybe it could be rephrased to explain how the two different paths work?

also, wasm compilation can be either wasm backend or asm2wasm, so wasm by itself implies both, but here we just mean the wasm backend, I think.